math/fibonacci.c

```diff
@@ -56,14 +56,19 @@ static void test()
 int main()
 {
     int number;
+    char buffer[8];
 
     // Performing the test
     test();
     printf("Tests passed...\n");
 
     // Getting n
     printf("Enter n to find nth fibonacci element: ");
-    scanf("%d", &number);
+    scanf("%2[0-9]s", buffer);
```

Comment:
That `s` in `%2[0-9]s` is excessive, `%2[0-9]` already specifies 2 digits. Also consider `12z`. This `scanf()` would parse `12` and store it into buffer, even though the input as a whole is not a valid number. There's also no code to repeat the request for input in case of invalid input. Here's a useful [link](https://devdocs.io/c/io/fscanf).

Reply:
What's the difference between keeping and not keeping 's' in the format specifier?

Reply:
> What's the difference between keeping and not keeping 's' in the format specifier?

Like I've said, `%2[0-9]` already is a format specification, so the `s` becomes a character literal, `%2[0-9]s` matches 2 digits plus a single `s`, like `20s`

Reply:
What can I have as the buffer size?